pageserver/src/tenant/timeline/walreceiver/walreceiver_connection.rs

```diff
@@ -343,6 +343,16 @@ pub(super) async fn handle_walreceiver_connection(
                             modification.commit(&ctx).await?;
                             uncommitted_records = 0;
                             filtered_records = 0;
+
+                            timeline
```

Comment:
This call helps mitigate the issue, but for our final version (https://github.com/neondatabase/neon/issues/6624) I'd like to build this into the timeline/writer, so that whenever we push values we are checking the size limit.  We could do that efficiently by tracking on the writer how many bytes it has written (and avoid taking the layermap lock to poll check_checkpoint_distance)